Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -1634,6 +1635,9 @@ void WebsiteDataStore::networkProcessDidTerminate(NetworkProcessProxy& networkPr
 {
     ASSERT(!m_networkProcess || m_networkProcess == &networkProcess);
     m_networkProcess = nullptr;
+
+    if (auto completionHandler = std::exchange(m_completionHandlerForRemovalFromNetworkProcess, { }))
```

Comment:
This seems inconvenient from the client standpoint no? It means they have to keep resetting their handler whenever the network process crashes? Seems this could easily be avoided on our side now that we store the completion handler of the data store instead of the NetworkProcessProxy? Any reason we don't?

Reply:
The purpose of this SPI is to let client know when they could operate (e.g. delete) on files that are being accessed by the network process, so as long as network process is not accessing the files -- either because network process is not running, or session is removed from network process, or network process gets killed -- the completionHandler will be called.

In the known use case, the client does destroy WebsiteDataStore first, so it's probably fine to only call it when WebsiteDataStore is destroyed. I will change it.

Reply:
My worry is that the network process will likely get relaunched right away if it just crashed. I don't even know if we notify the client that the network process relaunched.